earth_enterprise/src/fusion/autoingest/sysman/khSystemManager.cpp

```diff
@@ -72,20 +73,43 @@ khSystemManager::SetWantExit(void)
   }
 }
 
+// when SIGHUP is given, reload systemrc
+void systemrc_reload()
+{
+    Systemrc systemrc;
+    LoadSystemrc(systemrc);
+    uint32 logLevel = systemrc.logLevel;
+    assert(logLevel >= 0 && logLevel <= 7);
+    notify(NFY_WARN, "system log level changed to: %s",
+           khNotifyLevelToString(static_cast<khNotifyLevel>(logLevel)).c_str());
+    setNotifyLevel(static_cast<khNotifyLevel>(logLevel));
+}
+
+
+
 void
 khSystemManager::SignalLoop(void)
 {
   sigset_t waitset;
   sigemptyset(&waitset);
   sigaddset(&waitset, SIGINT);
   sigaddset(&waitset, SIGTERM);
-
+  sigaddset(&waitset, SIGHUP);
 
   int sig;
   sigwait(&waitset, &sig);
```

Comment:
It appears that this call never returns, even if I send a signal to the gesystemmanager. Somehow it's not picking up signals. It's not caused by your changes, but your changes won't work if it's not fixed.